_extensions/acronyms/acronyms_helpers.lua

```diff
@@ -104,5 +104,272 @@ function Helpers.str_to_boolean(value)
     return result
 end
 
+-- Parse limited markdown snippet (currently for italics etc.) using pandoc.read for robustness
+-- Returns list of Inlines
+function Helpers.parse_markdown_snippet(text)
+    local doc = pandoc.read(tostring(text) .. "\n", "markdown")
+    local blocks = doc.blocks
+    local inlines = {}
+    for _, b in ipairs(blocks) do
+        if b.t == "Para" or b.t == "Plain" then
+            for _, il in ipairs(b.c) do table.insert(inlines, il) end
+        end
+    end
+    return inlines
+end
+
+-- Determine whether a value (string, MetaInlines, Inlines or inline-array) contains
+-- user-supplied Markdown formatting (emphasis, strong, code, links, raw inline, etc.).
+function Helpers.contains_markdown(value)
+    if value == nil then return false end
+
+    -- If we have a Pandoc type, check AST nodes directly
+    if pandoc and pandoc.utils and pandoc.utils.type then
+        local typ = pandoc.utils.type(value)
+        if typ == "Inlines" or typ == "MetaInlines" or typ == "List" then
+            local inlines = Helpers.ensure_inlines(value)
+            for _, il in ipairs(inlines) do
+                if il.t ~= "Str" and il.t ~= "Space" and il.t ~= "SoftBreak" and il.t ~= "LineBreak" then
+                    return true
+                end
+            end
+            return false
+        end
+    end
+
+    -- If it's a plain Lua array of inline-like nodes, inspect element types
+    if type(value) == "table" then
+        for _, v in ipairs(value) do
+            if type(v) == "table" and v.t ~= nil then
+                if v.t ~= "Str" and v.t ~= "Space" and v.t ~= "SoftBreak" and v.t ~= "LineBreak" then
+                    return true
+                end
+            end
+        end
+    end
```

Comment:
This case does not return `false`, contrary to the previous case; is there a reason for that?
(note that this joins my previous comment; if both cases should return false, it would be clearer to merge them)